diesel_codegen_shared/src/schema_inference/pg.rs

```diff
@@ -171,3 +228,39 @@ fn skip_views() {
     assert!(table_names.contains(&"a_regular_table".to_string()));
     assert!(!table_names.contains(&"a_view".to_string()));
 }
+
+#[cfg(test)]
+mod tests {
+    use super::canonicalize_pg_type_name;
+    use super::camel_cased;
+
+    #[test]
+    fn camel_cased_empty() {
+        assert_eq!("", camel_cased(""));
+        assert_eq!("", camel_cased("_"));
+    }
+
+    #[test]
+    fn camel_cased_initial_caps() {
+        assert_eq!("Cased", camel_cased("CASED"));
+        assert_eq!("Cased", camel_cased("cased"));
+        assert_eq!("C", camel_cased("C"));
+        assert_eq!("C", camel_cased("c"));
+    }
+
+    #[test]
+    fn camel_cased_underscores() {
+        assert_eq!("CamelCased", camel_cased("camel_cased"));
+        assert_eq!("CamelCased", camel_cased("cAmEL_CAsEd"));
+        assert_eq!("CamelCased", camel_cased("camel_cased_"));
+        assert_eq!("CamelCased", camel_cased("_camel_cased"));
+        assert_eq!("CamelCased", camel_cased("_camel_cased_"));
+        assert_eq!("CamelCased", camel_cased("_camel__cased_"));
+    }
+
+    #[test]
+    fn camel_cased_i18n() {
+        assert_eq!("Außerdem", camel_cased("außerdem"));
+        assert_eq!("AuSSerdem", camel_cased("au_ßerdem"));
```

Comment:
In Rust, it's [an open issue](https://github.com/rust-lang/rust/issues/28979).

In Postgres, it appears perfectly valid to have both types and variants contain umlauts. I'll have to dig deeper into the documentation to see if there are any surprises with regards to other unusual characters.

It seems like eventual convergence (arbitrary whitespace-free Unicode allowed) is an eventual goal, but it would probably be better to check explicitly that we're generating a legal type name and die with a reasonable explanation rather than letting rustc barf.